devito/arch/compiler.py

```diff
@@ -487,24 +487,24 @@ def __init__(self, *args, **kwargs):
         if not configuration['safe-math']:
             self.cflags.append('-ffast-math')
 
+        language = kwargs.pop('language', configuration['language'])
         platform = kwargs.pop('platform', configuration['platform'])
 
-        if platform in [NVIDIAX, AMDGPUX]:
+        if platform is NVIDIAX:
             self.cflags.remove('-std=c99')
+        elif platform is AMDGPUX:
+            self.cflags.remove('-std=c99')
+            # Add flags for OpenMP offloading
+            if language in ['C', 'openmp']:
+                self.ldflags += ['-target', 'x86_64-pc-linux-gnu']
+                self.ldflags += ['-fopenmp']
+                self.ldflags += ['--offload-arch=%s' % platform.march]
         elif platform in [POWER8, POWER9]:
             # It doesn't make much sense to use AOMP on Power, but it should work
             self.cflags.append('-mcpu=native')
         else:
             self.cflags.append('-march=native')
 
-        # Generic amd flags
-        self.ldflags.extend(['-fopenmp', '-target', 'x86_64-pc-linux-gnu'])
```

Comment:
I know these can be removed but you still need the other flag `offload-arch`

Reply:
Also need to fix the arch detection to use `offload-arch` not `mygpu`

Reply:
ah OK, probably still using the old image?

Reply:
Unless you pushed a new one by hand yes it's still using the old one